extensions/sql-database-projects/src/models/project.ts

```diff
@@ -577,19 +402,141 @@ export class Project implements ISqlProject {
+
+	public async addPreDeploymentScript(relativePath: string): Promise<void> {
+		const result = await this.sqlProjService.addPreDeploymentScript(this.projectFileName, relativePath);
+		this.throwIfFailed(result);
+
+		await this.readPreDeployScripts();
+		await this.readNoneScripts();
+	}
+
+	public async deletePreDeploymentScript(relativePath: string): Promise<void> {
+		const result = await this.sqlProjService.deletePreDeploymentScript(this.projectFileName, relativePath);
+		this.throwIfFailed(result);
+
+		await this.readPreDeployScripts();
+	}
+
+	public async excludePreDeploymentScript(relativePath: string): Promise<void> {
+		const result = await this.sqlProjService.excludePreDeploymentScript(this.projectFileName, relativePath);
+		this.throwIfFailed(result);
+
+		await this.readPreDeployScripts();
+	}
+
+	//#endregion
+
+	//#region Post-deployment scripts
+
+	public async addPostDeploymentScript(relativePath: string): Promise<void> {
+		const result = await this.sqlProjService.addPreDeploymentScript(this.projectFileName, relativePath);
+		this.throwIfFailed(result);
+
+		await this.readPostDeployScripts();
+		await this.readNoneScripts();
+	}
+
+	public async deletePostDeploymentScript(relativePath: string): Promise<void> {
+		const result = await this.sqlProjService.deletePostDeploymentScript(this.projectFileName, relativePath);
+		this.throwIfFailed(result);
+
+		await this.readPostDeployScripts();
+	}
+
+	public async excludePostDeploymentScript(relativePath: string): Promise<void> {
+		const result = await this.sqlProjService.excludePostDeploymentScript(this.projectFileName, relativePath);
+		this.throwIfFailed(result);
+
+		await this.readPostDeployScripts();
+	}
+
+	//#endregion
+
+	//#region None items
+
+	public async addNoneItem(relativePath: string): Promise<void> {
+		const result = await this.sqlProjService.addNoneItem(this.projectFileName, relativePath);
+		this.throwIfFailed(result);
+
+		await this.readPostDeployScripts();
+		await this.readNoneScripts();
+	}
+
+	public async deleteNoneItem(relativePath: string): Promise<void> {
+		const result = await this.sqlProjService.deleteNoneItem(this.projectFileName, relativePath);
+		this.throwIfFailed(result);
+
+		await this.readPostDeployScripts();
+	}
+
+	public async excludeNoneItem(relativePath: string): Promise<void> {
+		const result = await this.sqlProjService.excludeNoneItem(this.projectFileName, relativePath);
+		this.throwIfFailed(result);
+
+		await this.readPostDeployScripts();
+	}
+
+	//#endregion
+
+	//#endregion
+
 	/**
 	 * Adds a folder to the project, and saves the project file
-	 *
+	 * TODO: delete once replaced with addFolder2
 	 * @param relativeFolderPath Relative path of the folder
 	 */
-	public async addFolderItem(relativeFolderPath: string): Promise<FileProjectEntry> {
-		const folderEntry = await this.ensureFolderItems(relativeFolderPath);
+	public async addFolder(relativeFolderPath: string): Promise<FileProjectEntry> {
```

Comment:
can this one be removed now?

Reply:
Should be fine.  sqldbproj.d.ts specifies the return type that shouldn't be necessary since it's basically just a typed wrapper for the argument.